library/core/src/num/mod.rs

```diff
@@ -999,15 +999,15 @@ macro_rules! impl_helper_for {
 }
 impl_helper_for! { i8 i16 i32 i64 i128 isize u8 u16 u32 u64 u128 usize }
 
-/// Determines if a string of text of that length of that radix could be guaranteed to be
-/// stored in the given type T.
+/// Determines length of text of a particular radix that could be guaranteed to be
+/// stored in the given type T without overflow.
 /// Note that if the radix is known to the compiler, it is just the check of digits.len that
 /// is done at runtime.
 #[doc(hidden)]
 #[inline(always)]
 #[unstable(issue = "none", feature = "std_internals")]
-pub fn can_not_overflow<T>(radix: u32, is_signed_ty: bool, digits: &[u8]) -> bool {
-    radix <= 16 && digits.len() <= mem::size_of::<T>() * 2 - is_signed_ty as usize
+pub fn safe_width<T>(radix: u32, is_signed_ty: bool) -> usize {
+    if radix > 16 { 0 } else { mem::size_of::<T>() * 2 - is_signed_ty as usize }
 }
```

Comment:
Do we still need `pub` here?

Reply:
Alas to test it. All the tests are integration tests.